### PR TITLE
Mobile menu

### PR DESCRIPTION
**Pull request by MarcoOquendoC**
# [Portfolio: mobile menu](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/7_mobile_menu.md)

## Hi good reviewer,

Penny

### These are the [changes](https://github.com/MarcoOquendoC/Portfolio/tree/mobile-menu) proposed to the [Portfolio](https://github.com/MarcoOquendoC/Portfolio) project:

[README.md]()
[index.html](https://github.com/MarcoOquendoC/Portfolio/blob/mobile-menu/index.html)
[styles.css](https://github.com/MarcoOquendoC/Portfolio/blob/mobile-menu/styles.css)
[script.js](https://github.com/MarcoOquendoC/Portfolio/blob/mobile-menu/script.js)

#### Project requirements
- Implement the mobile menu using JavaScript.
  - When the user clicks (or taps) the hamburger button, the mobile menu appears.
  - When the user clicks (or taps) the close (X) button, the mobile menu disappears.
  - When the user clicks (or taps) any of the mobile menu options, the mobile menu disappears.
  - When the user clicks (or taps) any of the mobile menu options, a correct part of the page is displayed.

- Replicate the design as much as possible using the info from [Figma template 3](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=48%3A988).
- Collaborate with coding partner [Daniel](https://github.com/dgonzalesi).

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  closeIcon.classList.add('hide');
+  about.classList.add('top-navigation');
+  contact.classList.add('top-navigation');
+  portfolio.classList.add('top-navigation');
+  nav.classList.remove('nav');
+  about.classList.remove('big-font');
+  contact.classList.remove('big-font');
+  portfolio.classList.remove('big-font');
+  closeIcon.classList.remove('close-icon');
+});
```
Comment (Bondok6) [inline script.js]: - [ ] Great job implementing the functionality of the `Menu` 👌💯, but I suggest following the [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md)
you've repeated the same lines of code in every function and It's not the best practice, To keep your code clean and [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) I suggest combining all the same lines of code in one function and using it 👌😉

> For Example 

```
const openMenu = () => {
  closeIcon.classList.remove('hide');
  nav.classList.add('nav');
  about.classList.add('big-font');
  contact.classList.add('big-font');
  portfolio.classList.add('big-font');
  closeIcon.classList.add('close-icon');
};

const closeMenu = () => {
  closeIcon.classList.add('hide');
  about.classList.add('top-navigation');
  contact.classList.add('top-navigation');
  portfolio.classList.add('top-navigation');
  nav.classList.remove('nav');
  about.classList.remove('big-font');
  contact.classList.remove('big-font');
  portfolio.classList.remove('big-font');
  closeIcon.classList.remove('close-icon');
};

hamburger.addEventListener('click', openMenu);

closeIcon.addEventListener('click', closeMenu);
portfolio.addEventListener('click', closeMenu);
about.addEventListener('click', closeMenu);
contact.addEventListener('click', closeMenu);
```

**Review by Bondok6 — CHANGES_REQUESTED** [review]
## Hi @MarcoOquendoC  and @dgonzalesi 🙋‍♂️, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- No Linter errors ✅
- Your repo is professional ✅
- Your project looks good ✅
- Great summary description ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MarcoOquendoC** [comment]
# Hi @Bondok6 ,
## Thank you for your review.

### The changes on the repository are listed below:

- Combine all the same lines of code in one function (it was obvious hehe thanks).
- *[Optional Aditional Requirements] Highlight nav word when scrolling.*

**Please let me know if the feature is well implemented.**

Have a good day bear

**Review by HAMMAS-SALEEM — APPROVED** [review]
# Status: Approved :green_circle:

Hey  @MarcoOquendoC :wave:

Great work on implementing the changes requested by the previous reviewer. Your project is complete. There is nothing else to say about your project. :+1:
Go ahead and merge it. :muscle:

## Highlights :1st_place_medal:
:heavy_check_mark: Following Github Flow
:heavy_check_mark: Following Linters
:heavy_check_mark: Your Menu is working as expected

### [Optional]
- [x] Your console window is showing some errors which are related to scrolling behavior. Please take these errors seriously because they can break your whole application :fearful:

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MarcoOquendoC** [comment]
Hi @HAMMAS-SALEEM,

I noticed the error and I didn't know what the problem was.

[OPTIONAL] It is already fixed with an if statement. :+1::slightly_smiling_face:

Thanks.

**Pull request merged**